app/gui/src/dashboard/layouts/AssetContextMenu.tsx

```diff
@@ -248,19 +251,31 @@ export default function AssetContextMenu(props: AssetContextMenuProps) {
               }}
             />
           )}
-        {asset.type === backendModule.AssetType.project && isCloud && (
+        {asset.type === backendModule.AssetType.project && isCloud && enableHybridExecution && (
           <ContextMenuEntry
             hidden={hidden}
             action="run"
             isDisabled={!canOpenProjects}
             tooltip={disabledTooltip}
-            doAction={() => {
-              openProjectMutation.mutate({
-                id: asset.id,
-                title: asset.title,
-                parentId: asset.parentId,
-                type: state.backend.type,
-                inBackground: true,
+            doAction={async () => {
+              invariant(localBackend != null, 'Local Backend is null')
+              const parentId = await remoteBackend.downloadProject(asset.id)
+              const assets = await localBackend.listDirectory({
```

Comment:
prefer `ensureQueryData` over calling methods manually: https://tanstack.com/query/latest/docs/reference/QueryClient#queryclientensurequerydata

Prefer mutations over direct calls here. Also, consider moving this out of the function for better reusability

Reply:
We generate a new id every time we download the project. We should not use cache here